analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -31,21 +31,19 @@
 using SonarAnalyzer.Extensions;
 using SonarAnalyzer.Helpers;
 using SonarAnalyzer.LiveVariableAnalysis.CSharp;
-using SonarAnalyzer.Rules.CSharp;
+using SonarAnalyzer.Rules.SymbolicExecution;
 using SonarAnalyzer.SymbolicExecution;
-using SonarAnalyzer.SymbolicExecution.Roslyn;
 using SonarAnalyzer.SymbolicExecution.Roslyn.Checks;
 using SonarAnalyzer.SymbolicExecution.Sonar;
 using StyleCop.Analyzers.Lightup;
 
-namespace SonarAnalyzer.Rules.SymbolicExecution
+namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed partial class SymbolicExecutionRunner : SonarDiagnosticAnalyzer
+    public sealed class SymbolicExecutionRunner : SymbolicExecutionRunnerBase
     {
-        private static readonly ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules = ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
-            .Add(LocksReleasedAllPaths.S2222, CreateFactory<LocksReleasedAllPaths>());
-        private static readonly ImmutableArray<ISymbolicExecutionAnalyzer> SonarRules = ImmutableArray.Create<ISymbolicExecutionAnalyzer>(    // ToDo: This should be migrated to AllRules
+        // ToDo: This should be migrated to SymbolicExecutionRunnerBase.AllRules. SymbolicExecutionRunnerBase.RegisterSupportedDiagnostics can be removed afterwards as well.
```

Comment:
The `issue` is actually MMF-2229 + MMF-2401 + MMF-2402 + MMF-2403 + MMF-2563 + MMF-2564. With each of them, few lines will disappear until it will remain empty.